pkg/services/provisioning/datasources/configreader/vcs_config_reader.go

```diff
@@ -0,0 +1,66 @@
+package configreader
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/provisioning/datasources"
+	"github.com/grafana/grafana/pkg/services/vcs"
+)
+
+type vcsConfigReader struct {
+	log log.Logger
+	vcs vcs.Service
+}
+
+func NewVCSConfigReader(log log.Logger, vcs vcs.Service) datasources.ConfigReader {
+	return &vcsConfigReader{log: log, vcs: vcs}
+}
+
+func (cr *vcsConfigReader) ReadConfigs(ctx context.Context) ([]*datasources.Configs, error) {
+	// Get all versioned datasources
+	datasourcesMap, err := cr.vcs.Latest(ctx, vcs.Datasource)
+	if err != nil {
+		return nil, err
+	}
+	if len(datasourcesMap) == 0 {
+		return []*datasources.Configs{}, nil
+	}
+
+	// Prepare a config for provisioning
+	cfg := datasources.Configs{
+		APIVersion:        1, // TODO manage to get this somehow?
```

Comment:
Maybe I could get it from config? But what happens if an end user actually uses different versions?